src/config/kubeconfig.js

```diff
@@ -1,9 +1,73 @@
 'use strict';
 
-const process = require('process');
 const pth = require('path');
 const fs = require('fs');
-const k8s = require('@kubernetes/client-node');
+// TODO: Uncomment after PR https://github.com/kubernetes-client/javascript/pull/748 lands
+// const k8s = require('@kubernetes/client-node');
+
+// TODO: Delete next two functions
+// after PR https://github.com/kubernetes-client/javascript/pull/748 lands
+
+function dropDuplicatesAndNils(a) {
+  return a.reduce((acceptedValues, currentValue) => {
+    // Good-enough algorithm for reducing a small (3 items at this point) array into an ordered list
+    // of unique non-empty strings.
+    if (currentValue && !acceptedValues.includes(currentValue)) {
+      return acceptedValues.concat(currentValue);
+    } else {
+      return acceptedValues;
+    }
+  }, []);
+}
+
+// Only public for testing.
```

Comment:
But we don't have any tests?

If we're trying to match the golang code, we should have pointers to the code we're matching.  Otherwise we should be using [`os.homedir()`](https://nodejs.org/docs/latest-v14.x/api/os.html#os_os_homedir) instead.

Reply:
Using `os.homedir()` would still not solve the gist of this issue, which is to favour directories containing `.kube/config` over others.

And I wrote unit tests for the upstream code. Including them here would be overkill. The comment was a stray from the copy-paste.